packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Package.swift

```diff
@@ -0,0 +1,46 @@
+// swift-tools-version: 5.9
+
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import PackageDescription
+
+let package = Package(
+  name: "video_player_avfoundation",
+  platforms: [
+    .iOS("12.0"),
+    .macOS("10.14"),
+  ],
+  products: [
+    .library(name: "video-player-avfoundation", targets: ["video_player_avfoundation"])
+  ],
+  dependencies: [],
+  targets: [
+    .target(
+      name: "video_player_avfoundation",
+      dependencies: [
+        .target(name: "video_player_avfoundation_ios", condition: .when(platforms: [.iOS])),
+        .target(name: "video_player_avfoundation_macos", condition: .when(platforms: [.macOS])),
+      ],
+      resources: [
+        .process("Resources")
+      ],
+      cSettings: [
+        .headerSearchPath("include/video_player_avfoundation")
+      ]
+    ),
+    .target(
```

Comment:
[Comment from previous PR](https://github.com/flutter/packages/pull/6634#discussion_r1589474756) by @stuartmorgan:
Interesting, so SPM doesn't allow source filtering without making separate targets?

Reply:
Correct, it doesn't allow you to conditionalize by platform very much, only [target dependencies](https://developer.apple.com/documentation/packagedescription/targetdependencycondition), [CSettings](https://developer.apple.com/documentation/packagedescription/csetting), [CXXSettings](https://developer.apple.com/documentation/packagedescription/cxxsetting), [SwiftSettings](https://developer.apple.com/documentation/packagedescription/swiftsetting), and [LinkerSettings](https://developer.apple.com/documentation/packagedescription/linkersetting).

[Sources](https://developer.apple.com/documentation/packagedescription/target/sources) can only be nil or an array of strings